mixin/thanos/alerts/ruler.libsonnet

```diff
@@ -0,0 +1,121 @@
+{
+  local thanos = self,
+  ruler+:: {
+    jobPrefix: error 'must provide job prefix for Thanos Ruler alerts',
+    selector: error 'must provide selector for Thanos Ruler alerts',
+  },
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'thanos-ruler.rules',
+        rules: [
+          {
+            alert: 'ThanosRulerQueueIsDroppingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to queue alerts.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_queue_alerts_dropped_total{%(selector)s}) > 0
+            ||| % thanos.ruler,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRulerSenderIsFailingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to send alerts to alertmanager.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_sender_alerts_dropped_total{%(selector)s}) > 0
+            ||| % thanos.ruler,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRulerHighRuleExaluationFailures',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to evaluate rules.',
+            },
+            expr: |||
+              (
+                sum by (job) (rate(prometheus_rule_evaluation_failures_total{%(selector)s}[5m]))
+              /
+                sum by (job) (rate(prometheus_rule_evaluations_total{%(selector)s}[5m]))
+              * 100 > 5
+              )
+            ||| % thanos.ruler,
+
+            'for': '5m',
+            labels: {
+              severity: 'warning',
+            },
+          },
+          {
+            alert: 'ThanosRulerHighRuleExaluationWarnings',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} has high number of evaluation warnings.',
+            },
+            expr: |||
+              sum by (job) (rate(thanos_rule_evaluation_with_warnings_total{%(selector)s}[5m])) > 0
+            ||| % thanos.ruler,
+
+            'for': '5m',
```

Comment:
5min seems pretty quick for this one. :thinking:

Reply:
15m? 30m?

Reply:
Changed to 15m.